### PR TITLE
Adding AnomalyMonitor and AnomalySubscription resources

### PR DESCRIPTION
**Pull request by araphn**
*Description of changes:*
Adding resources for Cost Anomaly Detection: AnomalyMonitor and AnomalySubscription

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: anomalymonitor/aws-ce-anomalymonitor.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        "DIMENSIONAL",
+        "CUSTOM"
+      ]
+    },
+    "MonitorName": {
+      "description": "The name of the monitor.",
+      "type": "string",
+      "pattern": "[\\S\\s]*",
+      "minLength": 0,
+      "maxLength": 1024
```
Comment (maoamzn) [inline anomalymonitor/aws-ce-anomalymonitor.json:28]: Do we need to have a max length of 1024? I think FE has a limit of 50. Not a blocker though.

Comment (araphn) [reply anomalymonitor/aws-ce-anomalymonitor.json:28]: We probably don't but I just want to be consistent with the public doc https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_AnomalyMonitor.html#awscostmanagement-Type-AnomalyMonitor-MonitorName

File: anomalymonitor/src/main/java/software/amazon/ce/anomalymonitor/CallbackContext.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package software.amazon.ce.anomalymonitor;
+
+import lombok.Builder;
+        import lombok.Data;
```
Comment (maoamzn) [inline anomalymonitor/src/main/java/software/amazon/ce/anomalymonitor/CallbackContext.java:4]: Extra spaces?

Comment (araphn) [reply anomalymonitor/src/main/java/software/amazon/ce/anomalymonitor/CallbackContext.java]: fixed!

**Review by maoamzn — APPROVED** [review]

**Review by tandonmt — APPROVED** [review]

File: anomalysubscription/aws-ce-anomalysubscription.json [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      "description": "Subscription ARN",
+      "type": "string",
+      "pattern": "^arn:aws[-a-z0-9]*:[a-z0-9]+:[-a-z0-9]*:[0-9]{12}:[-a-zA-Z0-9/:_]+$"
+    },
+    "Subscriber": {
+      "type": "object",
+      "properties": {
+        "Address": {
+          "type": "string",
+          "pattern": "(^[a-zA-Z0-9.!#$%&'*+=?^_\u2018{|}~-]+@[a-zA-Z0-9_-]+(\\.[a-zA-Z0-9_-]+)+$)|(^arn:(aws[a-zA-Z-]*):sns:[a-zA-Z0-9-]+:[0-9]{12}:[a-zA-Z0-9_-]+$)"
```
Comment (PatMyron) [inline anomalysubscription/aws-ce-anomalysubscription.json:16]: We're having issues with non-ASCII patterns in https://github.com/aws-cloudformation/cfn-python-lint/pull/1867#discussion_r562232974
JSON schema and CloudFormation [encourage using a minimal subset of regular expression syntax if possible](https://github.com/aws-cloudformation/cloudformation-cli/pull/675#discussion_r566298712)
(caught by running [`cfn validate`](https://github.com/aws-cloudformation/cloudformation-cli/pull/675))

**Pull request merged**